matrix-client-modes.el

```diff
@@ -44,12 +44,25 @@
                     "DEL "matrix-client-delete-backward-char
                     "M-v" matrix-client-scroll-down
                     "C-k" matrix-client-kill-line-or-unsent-message
+                    [remap indent-for-tab-command] matrix-client-tab
                     )))
     (cl-loop for (key fn) on mappings by #'cddr
-             do (define-key map (kbd key) fn))
+             do (define-key map (cl-typecase key
+                                  (string (kbd key))
+                                  (otherwise key))
+                  fn))
     map)
   "Keymap for `matrix-client-mode'.")
 
+(defun matrix-client-tab ()
+  "If point is before prompt, move point to prompt; otherwise call `indent-for-tab-command'."
+  (interactive)
+  (let ((prompt (matrix-client--prompt-position)))
+    (if (< (point) prompt)
+        (goto-char prompt)
+      (setq this-command #'indent-for-tab-command)
```

Comment:
Is this needed? I haven't really used `call-interactively` before, so I'm not sure.

Reply:
Apparently not.  I've never "forwarded" to `self-insert-command` before, so I copied that from another package, org-mode IIRC, but in this case it works without it, so I'll remove it.  Thanks.